src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.SymbolToProjectId.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal partial class SolutionState
```

Comment:
I'd prefer this to not be in a separate file, especially since it adds fields to SolutionState.

Reply:
that's atually why i want it a separate file.  i very much like when we can isolate fields and the operations on them in separate places.

Reply:
I don't like when it's not obvious what a particular class holds on in terms of state.

Reply:
Also, the rest of the class isn't split this way.

Reply:
If you want to isolate these then move them to a different type and reference that type from SolutionState.cs.

Reply:
Done.